wandb/sdk/launch/create_job.py

```diff
@@ -313,20 +313,36 @@ def _create_repo_metadata(
 
     # check if requirements.txt exists
     # start at the location of the python file and recurse up to the git root
-    req_dir = local_dir
-    while (
-        not os.path.exists(os.path.join(req_dir, "requirements.txt"))
-        and req_dir != tempdir
-    ):
-        req_dir = os.path.dirname(req_dir)
-
-    if not os.path.exists(os.path.join(req_dir, "requirements.txt")):
-        wandb.termerror(f"Could not find requirements.txt file in git repo at {path}")
-        return None
+    entrypoint_dir = os.path.dirname(entrypoint)
+    if entrypoint_dir:
+        req_dir = os.path.join(local_dir, entrypoint_dir)
+    else:
+        req_dir = local_dir
 
-    wandb.termlog(
-        f"Using requirements.txt in {req_dir.replace(tempdir, '') or 'repository root'}"
-    )
+    # If there is a Dockerfile.wandb in the starting rec dir, don't require a requirements.txt
+    if os.path.exists(os.path.join(req_dir, "Dockerfile.wandb")):
+        wandb.termlog(
+            f"Using Dockerfile.wandb in {req_dir.replace(tempdir, '') or 'repository root'}"
+        )
+    else:
+        while (
+            not os.path.exists(os.path.join(req_dir, "requirements.txt"))
+            and req_dir != tempdir
```

Comment:
I guess this is the base case intended to stop iteration, but can `entrypoint`/`entrypoint_dir` be a relative path that takes us out of `tempdir` to begin with? e.g., `../../../../../../fs_traversal_attack.py` ? We should be checking that absolute `req_dir` is prefixed by `tempdir`

Reply:
Good call I'm gonna do this